android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt

```diff
@@ -70,55 +70,50 @@ class AddonManager(
-                }
+            val featuredAddons = addonsProvider.getFeaturedAddons(allowCache, language = userLanguage)
+                .map { addon -> addon.filterTranslations(locales) }
                 .map { addon ->
                     installedExtensions[addon.id]?.let {
                         addon.copy(installedState = it.toInstalledState())
                     } ?: addon
                 }
 
-            val supportedAddonIds = supportedAddons.map { it.id }
-
-            // Get all installed addons that are not yet supported.
-            val unsupportedAddons = installedExtensions
-                .filterKeys { !supportedAddonIds.contains(it) }
+            // Build a list of installed extensions that are not built-in extensions AND not any of the featured
+            // extensions.
+            val featuredAddonIds = featuredAddons.map { it.id }
+            val otherInstalledExtensions = installedExtensions
+                .filterKeys { !featuredAddonIds.contains(it) }
                 .filterValues { !it.isBuiltIn() }
-                .map { extensionEntry ->
-                    val extension: WebExtension = extensionEntry.value
-                    val name = extension.getMetadata()?.name ?: extension.id
-                    val description = extension.getMetadata()?.description ?: extension.id
-
-                    // Temporary add-ons should be treated as supported
-                    val installedState = if (extension.getMetadata()?.temporary == true) {
-                        val icon = withContext(getIconDispatcher()) {
-                            extension.loadIcon(TEMPORARY_ADDON_ICON_SIZE)
-                        }
-                        extension.toInstalledState().copy(icon = icon)
-                    } else {
-                        extension.toInstalledState().copy(enabled = false, supported = false)
+
+            // We now want to build a list of add-on instances for these other installed extensions. We first retrieve
+            // the add-on GUIDs, and then we ask the add-ons provider to retrieve the metadata for these add-ons. After
+            // that, we add the installed state on each add-on instance.
+            //
+            // NB: We're keeping translations only for the default locale.
+            val installedAddonGUIDs = otherInstalledExtensions
+                .filterValues { it.getMetadata()?.temporary == false }
+                .map { extensionEntry -> extensionEntry.value.id }
+            val installedAddons = addonsProvider.getAddonsByGUIDs(installedAddonGUIDs, language = userLanguage)
+                .map { addon -> addon.filterTranslations(locales) }
+                .map { addon -> addon.copy(installedState = installedExtensions[addon.id]?.toInstalledState()) }
+            // Last but not least, we build the list of temporarily loaded extensions.
+            val temporarilyInstalledAddons = otherInstalledExtensions
```

Comment:
As we are going towards to general availability I think we may a good idea to revise if we still need temporarily extensions. No action needed now, just bring this up :)

Reply:
ah yes, we do because that's how developers can test their extension (e.g. `web-ext` loads add-ons temporarily)